src/main/scala/calc/Compiler.scala

```diff
@@ -3,6 +3,7 @@ package calc
 import org.scalajs.core.ir
 import ir.{Trees => irt, Types => irtpe}
 import ir.Definitions._
+import scala.collection.mutable.Map
```

Comment:
Using mutable stuff is a code smell. It is expected that this compiler be written in a purely functional style.